### PR TITLE
Add ecs_properties to job_definition of batch

### LINKED ISSUE #36034: [Enhancement]: AWS Batch `job_definition` ecs_properties parameter
### Description

AWS Batch job definition added a new parameter, ecs_properties

### Affected Resource(s) and/or Data Source(s)

* aws_batch_job_definition

### Potential Terraform Configuration

```terraform
resource "aws_batch_job_definition" "test" {
  name = "tf_test_batch_job_definition"
  type = "container"
  ecs_properties = {}
}
```

### References

- https://docs.aws.amazon.com/batch/latest/APIReference/API_EcsTaskProperties.html

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by takaishi**
### Description

Add ecs_properties for ECS resources of a job.

### Relations

Closes #36034

### Output from Acceptance Testing

```console
% AWS_PROFILE=private make testacc TESTS=TestAccBatchJobDefinition_ PKG=batch
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/batch/... -v -count 1 -parallel 20 -run='TestAccBatchJobDefinition_'  -timeout 360m
=== RUN   TestAccBatchJobDefinition_tags
=== PAUSE TestAccBatchJobDefinition_tags
=== RUN   TestAccBatchJobDefinition_tags_null
=== PAUSE TestAccBatchJobDefinition_tags_null
=== RUN   TestAccBatchJobDefinition_tags_AddOnUpdate
=== PAUSE TestAccBatchJobDefinition_tags_AddOnUpdate
=== RUN   TestAccBatchJobDefinition_tags_EmptyTag_OnCreate
=== PAUSE TestAccBatchJobDefinition_tags_EmptyTag_OnCreate
=== RUN   TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add
=== PAUSE TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add
=== RUN   TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace
=== PAUSE TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_providerOnly
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_providerOnly
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_overlapping
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_overlapping
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccBatchJobDefinition_tags_ComputedTag_OnCreate
=== PAUSE TestAccBatchJobDefinition_tags_ComputedTag_OnCreate
=== RUN   TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add
=== PAUSE TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add
=== RUN   TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace
=== PAUSE TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace
=== RUN   TestAccBatchJobDefinition_basic
=== PAUSE TestAccBatchJobDefinition_basic
=== RUN   TestAccBatchJobDefinition_attributes
=== PAUSE TestAccBatchJobDefinition_attributes
=== RUN   TestAccBatchJobDefinition_disappears
=== PAUSE TestAccBatchJobDefinition_disappears
=== RUN   TestAccBatchJobDefinition_PlatformCapabilities_ec2
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilities_ec2
=== RUN   TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
=== RUN   TestAccBatchJobDefinition_PlatformCapabilities_fargate
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilities_fargate
=== RUN   TestAccBatchJobDefinition_ContainerProperties_advanced
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_advanced
=== RUN   TestAccBatchJobDefinition_ContainerProperties_minorUpdate
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_minorUpdate
=== RUN   TestAccBatchJobDefinition_propagateTags
=== PAUSE TestAccBatchJobDefinition_propagateTags
=== RUN   TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== RUN   TestAccBatchJobDefinition_NodeProperties_basic
=== PAUSE TestAccBatchJobDefinition_NodeProperties_basic
=== RUN   TestAccBatchJobDefinition_NodeProperties_advanced
=== PAUSE TestAccBatchJobDefinition_NodeProperties_advanced
=== RUN   TestAccBatchJobDefinition_EKSProperties_basic
=== PAUSE TestAccBatchJobDefinition_EKSProperties_basic
=== RUN   TestAccBatchJobDefinition_EKSProperties_update
=== PAUSE TestAccBatchJobDefinition_EKSProperties_update
=== RUN   TestAccBatchJobDefinition_ECSProperties_basic
=== PAUSE TestAccBatchJobDefinition_ECSProperties_basic
=== RUN   TestAccBatchJobDefinition_ECSProperties_update
=== PAUSE TestAccBatchJobDefinition_ECSProperties_update
=== RUN   TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
=== PAUSE TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
=== RUN   TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== PAUSE TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== RUN   TestAccBatchJobDefinition_schedulingPriority
=== PAUSE TestAccBatchJobDefinition_schedulingPriority
=== CONT  TestAccBatchJobDefinition_tags
=== CONT  TestAccBatchJobDefinition_attributes
=== CONT  TestAccBatchJobDefinition_NodeProperties_basic
=== CONT  TestAccBatchJobDefinition_ECSProperties_update
=== CONT  TestAccBatchJobDefinition_EKSProperties_basic
=== CONT  TestAccBatchJobDefinition_NodeProperties_advanced
=== CONT  TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag
=== CONT  TestAccBatchJobDefinition_schedulingPriority
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly
=== CONT  TestAccBatchJobDefinition_ECSProperties_basic
=== CONT  TestAccBatchJobDefinition_basic
=== CONT  TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace
=== CONT  TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add
=== CONT  TestAccBatchJobDefinition_tags_ComputedTag_OnCreate
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag
=== CONT  TestAccBatchJobDefinition_ContainerProperties_advanced
=== CONT  TestAccBatchJobDefinition_EKSProperties_update
--- PASS: TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties (20.92s)
=== CONT  TestAccBatchJobDefinition_ContainerProperties_EmptyField
--- PASS: TestAccBatchJobDefinition_EKSProperties_basic (63.13s)
=== CONT  TestAccBatchJobDefinition_propagateTags
--- PASS: TestAccBatchJobDefinition_schedulingPriority (65.62s)
=== CONT  TestAccBatchJobDefinition_ContainerProperties_minorUpdate
--- PASS: TestAccBatchJobDefinition_NodeProperties_basic (67.64s)
=== CONT  TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag (68.44s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag (69.20s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_overlapping
--- PASS: TestAccBatchJobDefinition_ECSProperties_basic (72.81s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag (74.62s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_providerOnly
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag (74.91s)
=== CONT  TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
--- PASS: TestAccBatchJobDefinition_basic (79.78s)
=== CONT  TestAccBatchJobDefinition_tags_EmptyTag_OnCreate
--- PASS: TestAccBatchJobDefinition_ContainerProperties_EmptyField (60.42s)
=== CONT  TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add
--- PASS: TestAccBatchJobDefinition_tags_ComputedTag_OnCreate (84.00s)
=== CONT  TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
--- PASS: TestAccBatchJobDefinition_createTypeContainerWithNodeProperties (21.38s)
=== CONT  TestAccBatchJobDefinition_PlatformCapabilities_fargate
--- PASS: TestAccBatchJobDefinition_NodeProperties_advanced (109.80s)
=== CONT  TestAccBatchJobDefinition_PlatformCapabilities_ec2
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly (116.34s)
=== CONT  TestAccBatchJobDefinition_tags_AddOnUpdate
--- PASS: TestAccBatchJobDefinition_ECSProperties_update (116.70s)
=== CONT  TestAccBatchJobDefinition_tags_null
--- PASS: TestAccBatchJobDefinition_propagateTags (54.45s)
=== CONT  TestAccBatchJobDefinition_disappears
--- PASS: TestAccBatchJobDefinition_EKSProperties_update (119.82s)
--- PASS: TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace (121.10s)
--- PASS: TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add (124.27s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults (62.86s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_fargate (56.95s)
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_ec2 (48.66s)
--- PASS: TestAccBatchJobDefinition_disappears (41.61s)
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace (98.85s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly (98.12s)
--- PASS: TestAccBatchJobDefinition_attributes (172.58s)
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnCreate (94.60s)
--- PASS: TestAccBatchJobDefinition_tags_null (58.07s)
--- PASS: TestAccBatchJobDefinition_tags_AddOnUpdate (66.01s)
--- PASS: TestAccBatchJobDefinition_tags (186.79s)
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add (107.86s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_overlapping (124.12s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping (120.54s)
--- PASS: TestAccBatchJobDefinition_ContainerProperties_minorUpdate (131.87s)
--- PASS: TestAccBatchJobDefinition_ContainerProperties_advanced (202.00s)
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_providerOnly (133.64s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/batch      212.393s
```

### CODE REVIEW

File: website/docs/r/batch_job_definition.html.markdown [hunk 1]
```diff
@@ -134,6 +134,40 @@ resource "aws_batch_job_definition" "test" {
 }
 ```
 
+### Job Definitionn of type EKS
```
Comment (fabito) [inline website/docs/r/batch_job_definition.html.markdown:137]:
```suggestion
### Job Definition of type container using `ecs_properties`
```

I also think we should add examples using multicontainers definitions which is, to me, the key feature that was introduced.

https://docs.aws.amazon.com/batch/latest/userguide/multi-container-jobs-scenarios.html

File: internal/service/batch/job_definition_test.go [hunk 2]
```diff
@@ -1680,6 +1779,186 @@ resource "aws_batch_job_definition" "test" {
...
+}
+
+resource "aws_batch_job_definition" "test" {
+  name = %[1]q
+  type = "container"
+  platform_capabilities = ["FARGATE"]
+  ecs_properties {
+    task_properties {
+      execution_role_arn = aws_iam_role.ecs_task_execution_role.arn
+      containers {
```
Comment (fabito) [inline internal/service/batch/job_definition_test.go:1814]: It seems a bit odd to call it `containers` (plural) when the block defines a single container.
Shouldn't we call it `container` instead ?

File: website/docs/r/batch_job_definition.html.markdown [hunk 3]
```diff
@@ -256,6 +290,146 @@ The following arguments are optional:
 * `secret_name` - The name of the secret. The name must be allowed as a DNS subdomain name.
 * `optional` - (Optional) Specifies whether the secret or the secret's keys must be defined.
 
+### `ecs_properties`
+
+* `task_properties` - The properties of the Amazon ECS task definition on a job. See [`task_properties`](#task_properties) below.
+
+### `task_properties`
+
+* `containers` - The properties of the container. See [containers](#ecs_containers) below.
```
Comment (fabito) [inline website/docs/r/batch_job_definition.html.markdown:299]: This is actually a list of containers, right ?

File: website/docs/r/batch_job_definition.html.markdown [hunk 4]
```diff
@@ -256,6 +290,146 @@ The following arguments are optional:
...
+* `ephemeral_storage` - (Optional) The ephemeral storage settings for the task. See [ephemeral_storage](#ephemeral_storage) below.
+* `execution_role_arn` - (Optional) The Amazon Resource Name (ARN) of the IAM role that the AWS Batch can assume.
+* `ipc_mode` - (Optional) The IPC resource namespace to use for container in the task. The valid values are `host`, `task`, or `none`.
+* `network_configuration` - (Optional) The network configuration for the Amazon ECS task. See [network_configuration](#network_configuration) below.
+* `pic_mode` - (Optional) The process namespace to use for the container in the task. The valid values are `host` or `task`.
```
Comment (fabito) [inline website/docs/r/batch_job_definition.html.markdown:304]:
```suggestion
* `pid_mode` - (Optional) The process namespace to use for the container in the task. The valid values are `host` or `task`.
```

File: website/docs/r/batch_job_definition.html.markdown [hunk 5]
```diff
@@ -256,6 +290,146 @@ The following arguments are optional:
...
+* `platform_version` - (Optional) The platform version on which to run your task. If one is not specified, the LATEST platform version is used by default.
+* `runtime_platform` - (Optional) The compute environment architecture for AWS Batch job on Fargate. See [runtime_platform](#runtime_platform) below.
+* `task_role_arn` - The Amazon Resource Name (ARN) of the IAM role that the Amazon ECS task..
+* `volume` - (Optional) The volumes to mount on the container in the task. See [volume](#volume) below.
```
Comment (fabito) [inline website/docs/r/batch_job_definition.html.markdown:308]:
```suggestion
* `volumes` - (Optional) The volumes to mount on the container in the task. See [volume](#volume) below.
```

This is a list, right ?

Comment (takaishi) [reply website/docs/r/batch_job_definition.html.markdown]: Yes, fixed https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/4cf8e6e2edcfbc1eca7c836d862c8f924d67c118

File: website/docs/r/batch_job_definition.html.markdown (hunk 4 above)
Comment (takaishi) [reply website/docs/r/batch_job_definition.html.markdown]: Thank you! fixed https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/6a966d7f6a9f6e3c0bcad71d82e61e4cb22fe9fa

File: website/docs/r/batch_job_definition.html.markdown (hunk 1 above)
Comment (takaishi) [reply website/docs/r/batch_job_definition.html.markdown]: Agree, I added in https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/320085f325f146ae17fdc361c8a5e01de5ab8cee and https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/18ce938c07c3176c3a3dd20d0cde4e7c1937e745

File: internal/service/batch/job_definition_test.go (hunk 2 above)
Comment (takaishi) [reply internal/service/batch/job_definition_test.go]: I see. fix in https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/10a4475ef5711adc94412978f3fc428e4f641758

File: website/docs/r/batch_job_definition.html.markdown (hunk 3 above)
Comment (takaishi) [reply website/docs/r/batch_job_definition.html.markdown]: Yes, I fixed in https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/e90348c15176b284619438a3af227caec5ed34d4 and https://github.com/hashicorp/terraform-provider-aws/pull/37871/commits/10a4475ef5711adc94412978f3fc428e4f641758

File: website/docs/r/batch_job_definition.html.markdown [hunk 6]
```diff
@@ -134,6 +134,92 @@ resource "aws_batch_job_definition" "test" {
...
+### Job Definition of type container using `ecs_properties` with multiple containers
+
+```terraform
+resource "aws_batch_job_definition" "test" {
+  name = " tf_test_batch_job_definition_ecs"
+  type = "container"
+  ecs_properties {
+    task_properties {
+      host_network = true
+      containers {
```
Comment (fabito) [inline website/docs/r/batch_job_definition.html.markdown:180]:
```suggestion
      container {
```

File: website/docs/r/batch_job_definition.html.markdown [hunk 7]
```diff
@@ -134,6 +134,92 @@ resource "aws_batch_job_definition" "test" {
...
+### Job Definition of type container using `ecs_properties` with single container
+
+```terraform
+resource "aws_batch_job_definition" "test" {
+  name = " tf_test_batch_job_definition_ecs"
+  type = "container"
+  ecs_properties {
+    task_properties {
+      host_network = true
+      containers {
```
Comment (fabito) [inline website/docs/r/batch_job_definition.html.markdown:146]:
```suggestion
      container {
```

File: internal/service/batch/job_definition.go [hunk 8]
```diff
@@ -327,6 +327,419 @@ func ResourceJobDefinition() *schema.Resource {
...
+					Schema: map[string]*schema.Schema{
+						"task_properties": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"container": {
+										Type:     schema.TypeList,
+										Required: true,
```
Comment (fabito) [inline internal/service/batch/job_definition.go:345]:
```suggestion
										Required: true,
										MaxItems: 10,
```

According to the docs: 
> You can include from one (1) to ten (10) containers in the EcsProperties structure. For more information see [EcsTaskProperties:containers](https://docs.aws.amazon.com/batch/latest/APIReference/API_EcsTaskProperties.html).
https://docs.aws.amazon.com/batch/latest/userguide/multi-container-jobs.html#multi-container-general

**Comment by davids-flwls** [comment]
This is something our team is having to work around, by building local docker agents in our deployed images, and we'd love to be able to do this via TF instead. 

Is there anything I can do to help get this change over the line?

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccBatchJobDefinition_' PKG=batch ACCTEST_PARALLELISM=3
\make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/batch/... -v -count 1 -parallel 3  -run=TestAccBatchJobDefinition_ -timeout 360m
=== RUN   TestAccBatchJobDefinition_tags
=== PAUSE TestAccBatchJobDefinition_tags
=== RUN   TestAccBatchJobDefinition_tags_null
=== PAUSE TestAccBatchJobDefinition_tags_null
=== RUN   TestAccBatchJobDefinition_tags_AddOnUpdate
=== PAUSE TestAccBatchJobDefinition_tags_AddOnUpdate
=== RUN   TestAccBatchJobDefinition_tags_EmptyTag_OnCreate
=== PAUSE TestAccBatchJobDefinition_tags_EmptyTag_OnCreate
=== RUN   TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add
=== PAUSE TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add
=== RUN   TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace
=== PAUSE TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_providerOnly
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_providerOnly
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_overlapping
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_overlapping
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccBatchJobDefinition_tags_ComputedTag_OnCreate
=== PAUSE TestAccBatchJobDefinition_tags_ComputedTag_OnCreate
=== RUN   TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add
=== PAUSE TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add
=== RUN   TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace
=== PAUSE TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace
=== RUN   TestAccBatchJobDefinition_basic
=== PAUSE TestAccBatchJobDefinition_basic
=== RUN   TestAccBatchJobDefinition_attributes
=== PAUSE TestAccBatchJobDefinition_attributes
=== RUN   TestAccBatchJobDefinition_disappears
=== PAUSE TestAccBatchJobDefinition_disappears
=== RUN   TestAccBatchJobDefinition_PlatformCapabilities_ec2
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilities_ec2
=== RUN   TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
=== RUN   TestAccBatchJobDefinition_PlatformCapabilities_fargate
=== PAUSE TestAccBatchJobDefinition_PlatformCapabilities_fargate
=== RUN   TestAccBatchJobDefinition_ContainerProperties_advanced
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_advanced
=== RUN   TestAccBatchJobDefinition_ContainerProperties_minorUpdate
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_minorUpdate
=== RUN   TestAccBatchJobDefinition_propagateTags
=== PAUSE TestAccBatchJobDefinition_propagateTags
=== RUN   TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== PAUSE TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== RUN   TestAccBatchJobDefinition_NodeProperties_basic
=== PAUSE TestAccBatchJobDefinition_NodeProperties_basic
=== RUN   TestAccBatchJobDefinition_NodeProperties_advanced
=== PAUSE TestAccBatchJobDefinition_NodeProperties_advanced
=== RUN   TestAccBatchJobDefinition_EKSProperties_basic
=== PAUSE TestAccBatchJobDefinition_EKSProperties_basic
=== RUN   TestAccBatchJobDefinition_EKSProperties_update
=== PAUSE TestAccBatchJobDefinition_EKSProperties_update
=== RUN   TestAccBatchJobDefinition_EKSProperties_imagePullSecrets
=== PAUSE TestAccBatchJobDefinition_EKSProperties_imagePullSecrets
=== RUN   TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
=== PAUSE TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
=== RUN   TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== PAUSE TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
=== RUN   TestAccBatchJobDefinition_schedulingPriority
=== PAUSE TestAccBatchJobDefinition_schedulingPriority
=== RUN   TestAccBatchJobDefinition_emptyRetryStrategy
=== PAUSE TestAccBatchJobDefinition_emptyRetryStrategy
=== RUN   TestAccBatchJobDefinition_ECSProperties_update
=== PAUSE TestAccBatchJobDefinition_ECSProperties_update
=== CONT  TestAccBatchJobDefinition_tags
=== CONT  TestAccBatchJobDefinition_ContainerProperties_EmptyField
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag
--- PASS: TestAccBatchJobDefinition_ContainerProperties_EmptyField (11.61s)
=== CONT  TestAccBatchJobDefinition_createTypeContainerWithNodeProperties
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nullNonOverlappingResourceTag (12.59s)
=== CONT  TestAccBatchJobDefinition_ECSProperties_update
--- PASS: TestAccBatchJobDefinition_createTypeContainerWithNodeProperties (2.77s)
=== CONT  TestAccBatchJobDefinition_emptyRetryStrategy
--- PASS: TestAccBatchJobDefinition_emptyRetryStrategy (9.61s)
=== CONT  TestAccBatchJobDefinition_schedulingPriority
--- PASS: TestAccBatchJobDefinition_ECSProperties_update (16.52s)
=== CONT  TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties
--- PASS: TestAccBatchJobDefinition_createTypeMultiNodeWithContainerProperties (2.94s)
=== CONT  TestAccBatchJobDefinition_EKSProperties_basic
--- PASS: TestAccBatchJobDefinition_schedulingPriority (10.81s)
=== CONT  TestAccBatchJobDefinition_EKSProperties_imagePullSecrets
--- PASS: TestAccBatchJobDefinition_tags (40.39s)
=== CONT  TestAccBatchJobDefinition_EKSProperties_update
--- PASS: TestAccBatchJobDefinition_EKSProperties_basic (11.01s)
=== CONT  TestAccBatchJobDefinition_PlatformCapabilities_ec2
--- PASS: TestAccBatchJobDefinition_EKSProperties_imagePullSecrets (10.84s)
=== CONT  TestAccBatchJobDefinition_propagateTags
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_ec2 (10.65s)
=== CONT  TestAccBatchJobDefinition_ContainerProperties_minorUpdate
--- PASS: TestAccBatchJobDefinition_propagateTags (8.75s)
=== CONT  TestAccBatchJobDefinition_ContainerProperties_advanced
--- PASS: TestAccBatchJobDefinition_EKSProperties_update (17.53s)
=== CONT  TestAccBatchJobDefinition_PlatformCapabilities_fargate
--- PASS: TestAccBatchJobDefinition_PlatformCapabilities_fargate (11.52s)
=== CONT  TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults
--- PASS: TestAccBatchJobDefinition_PlatformCapabilitiesFargate_containerPropertiesDefaults (11.72s)
=== CONT  TestAccBatchJobDefinition_NodeProperties_advanced
--- PASS: TestAccBatchJobDefinition_NodeProperties_advanced (18.02s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping
--- PASS: TestAccBatchJobDefinition_ContainerProperties_advanced (48.00s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nullOverlappingResourceTag (13.13s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag
--- PASS: TestAccBatchJobDefinition_ContainerProperties_minorUpdate (69.15s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_emptyProviderOnlyTag (14.55s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_nonOverlapping (38.59s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_emptyResourceTag (16.12s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_overlapping
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_updateToResourceOnly (23.48s)
=== CONT  TestAccBatchJobDefinition_basic
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_updateToProviderOnly (25.64s)
=== CONT  TestAccBatchJobDefinition_disappears
--- PASS: TestAccBatchJobDefinition_basic (12.62s)
=== CONT  TestAccBatchJobDefinition_attributes
--- PASS: TestAccBatchJobDefinition_disappears (9.22s)
=== CONT  TestAccBatchJobDefinition_NodeProperties_basic
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_overlapping (36.86s)
=== CONT  TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add
--- PASS: TestAccBatchJobDefinition_NodeProperties_basic (13.67s)
=== CONT  TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace
--- PASS: TestAccBatchJobDefinition_attributes (31.14s)
=== CONT  TestAccBatchJobDefinition_tags_ComputedTag_OnCreate
--- PASS: TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Add (22.66s)
=== CONT  TestAccBatchJobDefinition_tags_DefaultTags_providerOnly
--- PASS: TestAccBatchJobDefinition_tags_ComputedTag_OnUpdate_Replace (22.72s)
=== CONT  TestAccBatchJobDefinition_tags_AddOnUpdate
--- PASS: TestAccBatchJobDefinition_tags_ComputedTag_OnCreate (15.29s)
=== CONT  TestAccBatchJobDefinition_tags_EmptyTag_OnCreate
--- PASS: TestAccBatchJobDefinition_tags_AddOnUpdate (19.38s)
=== CONT  TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnCreate (21.78s)
=== CONT  TestAccBatchJobDefinition_tags_null
--- PASS: TestAccBatchJobDefinition_tags_DefaultTags_providerOnly (41.92s)
=== CONT  TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Replace (19.14s)
--- PASS: TestAccBatchJobDefinition_tags_null (14.57s)
--- PASS: TestAccBatchJobDefinition_tags_EmptyTag_OnUpdate_Add (28.60s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/batch	274.227s
```

**Comment by ewbankkit** [comment]
@takaishi Thanks for the contribution :tada: :clap:.
To match `container_properties` and `node_properties` (and the `aws_ecs_task_definition` resource's `container_properties` attribute), I changed `ecs_properties` to a JSON string.

**Pull request merged**